docs/ARM.md

```diff
@@ -165,14 +165,26 @@ knife azurerm server create
   -c ~/.chef/knife.rb
 ```
 
-User can use existing virtual network and subnet while server create by providing `--azure-vnet-name` and `--azure-vnet-subnet-name` options.
+User can also pass names for virtual network and subnet while server create by providing `--azure-vnet-name` and `--azure-vnet-subnet-name` options. Following are the possible scenarios for the usage of these two options:
 
-Note:
+1. `--azure-vnet-name` and `--azure-vnet-subnet-name` options not provided by user: `--azure-vm-name` will be taken as the default value for both the options.
 
-1. If these options are not given, default vnet and subnet with the VM name will be created.
-2. User needs to provide valid existing vnet name and subnet name otherwise it will raise vnet or subnet not found error.
-3. Vnet should belong to the same resource group that is provided in the command.
-4. Subnet should belong to the same vnet that is provided in the command.
+2. `--azure-vnet-name` and `--azure-vnet-subnet-name` options provided by user: User provided values will be assigned to these options respectively.
+
+3. `--azure-vnet-name` option provided by user but `--azure-vnet-subnet-name` option not provided by user: `--azure-vm-name` will be assigned as the default value to the `--azure-vnet-subnet-name` option.
+
+4. `--azure-vnet-subnet-name` option provided by user but `--azure-vnet-name` option not provided by user: It will raise error saying `--azure-vnet-name` option must also be specified with the `--azure-vnet-subnet-name` option.
+
+
+***Note***:
+
+- Vnet and subnet (user provided or default valued) if do not exist will be newly created in the resource group while server creation.
+- Vnet and subnet (user provided or default valued) if exist will be used while server creation.
+- Vnet (user provided or default valued) if exist and subnet (user provided or default valued) do not exist then new subnet will be added in the virtual network.
```

Comment:
This can be simplified as: 
```
If `Vnet` exists and `subnet` does not exist, then a new `subnet` will be added in the virtual network.
```
To avoid duplication of `(user provided or default valued)`, we can have one more point in the beginning saying `Vnet and subnet can user provided or default`